File: src/main/java/me/oriharel/machinery/data/Range.java
package me.oriharel.machinery.data;

import java.util.Objects;
import java.util.concurrent.ThreadLocalRandom;

public class Range {
    private final int start;
    private final int end;

    public Range(int start, int end) {
        this.start = start;
        this.end = end;
    }

    public int random() {
        return ThreadLocalRandom.current().nextInt(start, end + 1);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Range range = (Range) o;
        return start == range.start &&
                end == range.end;
    }

    @Override
    public int hashCode() {
        return Objects.hash(start, end);
    }

    public int getStart() {
        return start;
    }

    public int getEnd() {
        return end;
    }
}

File: src/main/kotlin/me/oriharel/machinery/data/Range.kt
package me.oriharel.machinery.data

import java.util.*
import java.util.concurrent.ThreadLocalRandom

class Range(private val start: Int, private val end: Int) {
    fun random(): Int {
        return ThreadLocalRandom.current().nextInt(start, end + 1)
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val range = o as Range
        return start == range.start &&
                end == range.end
    }

    override fun hashCode(): Int {
        return Objects.hash(start, end)
    }

}